jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPartFormData.java

```diff
@@ -468,23 +475,29 @@ public void onPartContent(Content.Chunk chunk)
                             onFailure(x);
                         }
 
-                        partChunks.forEach(Content.Chunk::release);
-                        partChunks.clear();
+                        try (AutoLock ignored = lock.lock())
+                        {
+                            partChunks.forEach(Content.Chunk::release);
+                            partChunks.clear();
+                        }
                         return;
                     }
                 }
             }
             // Retain the chunk because it is stored for later use.
             chunk.retain();
-            partChunks.add(chunk);
+            try (AutoLock ignored = lock.lock())
+            {
+                partChunks.add(chunk);
+            }
         }
 
         private void write(ByteBuffer buffer) throws Exception
         {
             int remaining = buffer.remaining();
             while (remaining > 0)
             {
-                int written = fileChannel.write(buffer);
+                int written = fileChannel().write(buffer);
```

Comment:
`fileChannel()` may return `null`.

Reply:
Sure, but apart from throwing a different type of exception than NPE, nothing different should be done.